### DISCUSSION TITLE
Time point at time 0 added by default in R

### DISCUSSION THREAD

**Discussion by teutonicod** [Open-Systems-Pharmacology/Forum#1651] [Questions & Problems] [answered]
Dear all,

I want to simulate in R a series of timepoint corresponding to observations. 
To do that, I take the real time measured for each patient and set the time in the simulation to be run. After I run the simulation, the timepoint at time 0 is included in the output, also if it was not in my original time vectors. Is there a way to avoid this? I need to create an automatic function to do that, and since sometime the time=0 is indeed present, I cannot just remove it systematically. An example of the code I use is the following:

  ```
mySimulation$outputSchema$clear()
  myTimes <- c(1,2,4,6,10)
  myTimes <- myTimes*60
  mySimulation$outputSchema$addTimePoints(myTimes)
```

Is there a different code or option to avoid that the time=0 is included by default in the output of the simulation?

Thanks a lot for your help!

Donato

**Comment by msevestre** [comment]
Hi @teutonicod ,

We do not add 0 automatically. Do you have an application that starts at t=0 maybe?

**Reply by teutonicod** [reply]
Hi @msevestre,

Thanks for your reply! Yes, indeed I have an oral administration that is done at t=0. Does this mean that this timepoint will be added to the simulation also if 0 is not part of the timepoints of the simulation? 
If this is the case, is there a path to the administration parameter to get the administration time in R?

**Reply by msevestre** [reply, marked as answer]
Why don't you want to simulate when the administration is taking place? This does not make sense to me. If your apply at time and 0 and your simulation starts at 2, well you have no application at all. Something I am not quite getting. 
Yes you can access the parameter in R as you access any other parameters. (using wildcards etc..) see the discussion here
https://github.com/Open-Systems-Pharmacology/Forum/discussions/1644#discussioncomment-7342949

**Reply by teutonicod** [reply]
Thanks a lot, I will have a look! I am triyng to get a simulation that I want to compare with observed data, so indipendently from when the administration take place, if I don't have an observation at time 0 I don't want to have a simulation for that time point.
Thanks for your help!

**Reply by msevestre** [reply]
IF your simulation starts AFTER The administration is taking place, the administration won't be taken into account. That's why the point is added. If your goal is just to remove the point at 0, this is something you can do as post processing in R. There might be another way of doing this 

To be honest, I am not sure what happens if you only use time points and no continuous interval. The idea is to have time points that you want to meet explicitly.

**Reply by StephanSchaller** [reply]
As Michael said, a solution would be postprocessing the full simulation output with an interpolation function in R using, e.g., [approx](https://stat.ethz.ch/R-manual/R-devel/library/stats/html/approxfun.html) and your data vector.

**Reply by teutonicod** [reply]
Thanks a lot! It is clear now what Pk-Sim does!
